tiledb/sm/query/readers/result_tile.h

```diff
@@ -71,12 +71,61 @@ class Subarray;
 class ResultTile {
  public:
   /**
-   * For each fixed-sized attributes, the second tile in the tuple is ignored.
-   * For var-sized attributes, the first tile is the offsets tile and the second
-   * tile is the var-sized values tile. If the attribute is nullable, the third
-   * tile contains the validity vector.
+   * Class definition for the tile tuple.
    */
-  typedef tuple<Tile, Tile, Tile> TileTuple;
+  class TileTuple {
+   public:
+    /* ********************************* */
+    /*     CONSTRUCTORS & DESTRUCTORS    */
+    /* ********************************* */
+
+    /** Default constructor. */
+    TileTuple() = default;
+
+    /* ********************************* */
+    /*                API                */
+    /* ********************************* */
+
+    /** @returns Fixed tile. */
+    Tile& fixed_tile() {
+      return fixed_tile_;
+    }
+
+    /** @returns Var tile. */
+    Tile& var_tile() {
+      return var_tile_;
+    }
+
+    /** @returns Validity tile. */
+    Tile& validity_tile() {
+      return validity_tile_;
+    }
+
+    /** @returns Fixed tile. */
+    const Tile& fixed_tile() const {
+      return fixed_tile_;
+    }
+
+    /** @returns Var tile. */
+    const Tile& var_tile() const {
+      return var_tile_;
+    }
+
+    /** @returns Validity tile. */
+    const Tile& validity_tile() const {
+      return validity_tile_;
+    }
+
+   private:
+    /** Stores the fixed data tile. */
+    Tile fixed_tile_;
+
+    /** Stores the var data tile. */
```

Comment:
Semantically, var and validity are optional, right? So consider making that concrete here as `std::optional<Tile>` (doesn't need to change the accessor interface).

Reply:
Correct. I was initially worried about perf and was going to do this in a later change, but your comment pushed me to measure the difference today and I think we are good.